lib/test/reporter/tap.js

```diff
@@ -22,113 +23,77 @@ let testModule; // Lazy loaded due to circular dependency.
 
 function lazyLoadTest() {
   testModule ??= require('internal/test_runner/test');
-
   return testModule;
 }
 
-class TapStream extends Readable {
-  #buffer;
-  #canPush;
-
-  constructor() {
-    super();
-    this.#buffer = [];
-    this.#canPush = true;
-  }
 
-  _read() {
-    this.#canPush = true;
-
-    while (this.#buffer.length > 0) {
-      const line = ArrayPrototypeShift(this.#buffer);
-
-      if (!this.#tryPush(line)) {
-        return;
+async function * tapReporter(source) {
+  for await (const { type, data } of source) {
+    switch (type) {
+      case 'test:fail':
+        yield reportTest(data.nesting, data.testNumber, 'not ok', data.name, data.directive);
+        yield reportDetails(data.nesting, data.details);
```

Comment:
Should we be yielding twice here? Same question for `'test:pass'`.

Reply:
just kept the same behavior as before, we sent this as two different chunks